aggoracle/chaingerreader/evm.go

```diff
@@ -73,7 +83,11 @@ func (e *EVMChainGERReader) GetInjectedGERsForRange(ctx context.Context,
 
 		eventBlockNum := iter.Event.Raw.BlockNumber
 
-		injectedGERs[eventBlockNum] = append(injectedGERs[eventBlockNum], iter.Event.NewGlobalExitRoot)
+		injectedGERs[eventBlockNum] = append(injectedGERs[eventBlockNum], InjectedGER{
+			BlockNumber:    eventBlockNum,
+			BlockIndex:     iter.Event.Raw.Index,
+			GlobalExitRoot: iter.Event.NewGlobalExitRoot,
+		})
 	}
 
```

Comment:
IMO we should handle the UpdateHashChainRemoval events and return only those GERs that are not removed in a meantime (https://github.com/agglayer/agg-contracts-internal/blob/feature/ongoing-v0.3.0/contracts/v2/sovereignChains/GlobalExitRootManagerL2SovereignChain.sol#L48-L51).

Reply:
Agreed. I will add that logic.

Reply:
Done: https://github.com/agglayer/aggkit/pull/377/commits/689daf2ad01a379704262ac491e30292843344d5